### PR TITLE
Replace AsyncClient with AsyncZyteAPI

### PR DESCRIPTION
**Pull request by Gallaecio**

### CODE REVIEW

**Review by wRAR — APPROVED** [review]

File: docs/asyncio_api.rst [hunk 1]
```diff
@@ -4,47 +4,44 @@
...
-        client = AsyncClient(n_conn=n_conn)
+        client = AsyncZyteAPI(n_conn=n_conn)
         requests = [
             {"url": url, "browserHtml": True}
             for url in urls
         ]
         async with create_session(n_conn) as session:
-            res_iter = client.request_parallel_as_completed(requests, session=session)
+            res_iter = client.iter(requests, session=session)
             for fut in res_iter:
```
Comment (kmike) [inline docs/asyncio_api.rst:45]: Does the order in res_iter match the order or items in `requests`? Should we document it?

File: zyte_api/_async.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        self,
+        queries: List[dict],
+        *,
+        endpoint: str = 'extract',
+        session: Optional[aiohttp.ClientSession] = None,
+        handle_retries=True,
+        retrying: Optional[AsyncRetrying] = None,
+    ) -> AsyncGenerator[Union[dict, Exception], None]:
+        """ Send multiple requests to Zyte API in parallel.
+        Return an `asyncio.as_completed` iterator.
```
Comment (kmike) [inline zyte_api/_async.py:128]: Should we document the behavior for the exceptions?

It also looks like the example in the docs handles exceptions in a different way (via try/except), while the new code yields the exceptions. What's the advantage of yielding the exceptions here?

Comment (kmike) [reply zyte_api/_async.py:128]: I think no (see the example in the current docs). I think the the difference with the sync version is that we return an iterable of awaitables, and the exception happens when user awaits the awaitable - you can catch it in the code. In the sync version there is no awaitable to await, so it stops the iteration.

File: docs/asyncio_api.rst (hunk 1 above)
Comment (Gallaecio) [reply docs/asyncio_api.rst:45]: eafd59f

File: zyte_api/aio/client.py [hunk 3]
```diff
@@ -19,6 +20,16 @@
...
 
 
+warn(
+    (
+        "The zyte_api.aio.client module is deprecated. Replace AsyncClient "
+        "with zyte_api.AsyncZyteAPI (note that method names are different) "
+        "and create_session with zyte_api.create_session."
+    ),
+    DeprecationWarning,
+)
```
[the remark this answers is not in the document]
Comment (Gallaecio) [reply zyte_api/aio/client.py:32]: 95b85e7

File: docs/asyncio_api.rst [hunk 4]
```diff
@@ -4,47 +4,44 @@
 asyncio API
 ===========
 
-Create an instance of the ``AsyncClient`` to use the asyncio client API.
+Create an instance of the ``AsyncZyteAPI`` to use the asyncio client API.
 You can use the method ``request_raw`` to perform individual requests:
```
Comment (lopuhin) [inline docs/asyncio_api.rst:8]: @Gallaecio just noticed that this is referencing the old method name

Comment (Gallaecio) [reply docs/asyncio_api.rst:8]: Will fix in a follow-up PR. There are also a few parts in the code that still point to AsyncClient I think.

**Pull request merged**